internal/db/legacy_adapter.go

```diff
@@ -0,0 +1,146 @@
+package db
```

Comment:
why do we need this legacy adapter?

Reply:
moving to refactor PR where it used for testing during WIP